src/actinia_core/resources/process_chain_monitoring.py

```diff
@@ -0,0 +1,491 @@
+    plt.xticks(np.arange(1, len(y) + 1, step=1))
+    plt.savefig(result_file)
+    return result_file
+
+
+def compute_mapset_size_diffs(mapset_sizes):
+    diffs = [None] * len(mapset_sizes)
+    diffs[0] = mapset_sizes[0] # TODO not correct cause an empty mapset has also a size!
+    for i in range(1, len(mapset_sizes)):
+        diffs[i] = mapset_sizes[i] - mapset_sizes[i-1]
+    return diffs
+
+
+class MapsetSizeResponseModel(Schema):
+    """Response schema for mapset sizes of a resource
+    """
+    type = 'object'
+    properties = {
+        'status': {
+            'type': 'string',
+            'description': 'The status of the resource, values: success, error'
+        },
+        'mapset_sizes': {
+            'type': 'array',
+            'items': {"type": "integer"},
+            'description': 'The list of mapset sizes of a resource in bytes'
+        }
+    }
+    example = {"mapset_sizes": [29946, 29946], "status": "success"}
+    required = ["status", "mapset_sizes"]
+
+
+class MapsetSizeResource(ResourceManager):
+    """
+    This class returns the mapset sizes of a resource
+    """
+    def __init__(self):
+
+        # Configuration
+        ResourceManager.__init__(self)
+
+    @swagger.doc({
+        'tags': ['Process Chain Monitoring'],
+        'description': 'Get the mapset sizes of a resource. Minimum required user role: user.',
+        'parameters': [
+            {
+                'name': 'user_id',
+                'description': 'The unique user name/id',
+                'required': True,
+                'in': 'path',
+                'type': 'string'
+            },
+            {
+                'name': 'resource_id',
+                'description': 'The id of the resource',
+                'required': True,
+                'in': 'path',
+                'type': 'string'
+            }
+        ],
+        'responses': {
+            '200': {
+                'description': 'The current state of the resource',
+                'schema': MapsetSizeResponseModel
+            },
+            '400': {
+                'description': 'The error message if the resource does not exists',
+                'schema': SimpleResponseModel
+            }
+        }
+    })
+    def get(self, user_id, resource_id):
+        """Get the sizes of mapset of a resource."""
+
+        ret = self.check_permissions(user_id=user_id)
+        if ret:
+            return ret
+
+        response_data = self.resource_logger.get(user_id, resource_id)
+
+        if response_data is not None:
+            http_code, pc_response_model = pickle.loads(response_data)
+
+            pc_status = pc_response_model['status']
+            if pc_status in ['accepted', 'running']:
+                return make_response(jsonify(SimpleResponseModel(
+                    status="error",
+                    message="Resource is not ready it is %s" % pc_status)),
+                    400)
+
+            mapset_sizes = [proc['mapset_size']
```

Comment:
Would it somehow be possible to add the step and/or module id to the size information? It would be more easy to analyze. Doesn't need to be done in this PR but would be a nice enhancement on top of the cherry on the cake.

Reply:
The module id is not in the process_log, so it needs to be added or taken from the process chain